server/apps/io/reuters.py

```diff
@@ -102,9 +102,12 @@ def get_ids(self, channel, last_updated, updated):
         ids = []
         payload = {'channel': channel, 'fieldsRef': 'id'}
         payload['dateRange'] = "%s-%s" % (self.format_date(last_updated), self.format_date(updated))
+        logger.info('Reuters requesting Date Range |{}| for channel {}'.format(payload['dateRange'], channel))
         tree = self.get_tree('items', payload)
         for result in tree.findall('result'):
-            ids.append(result.find('guid').text)
+            id = result.find('guid').text
+            if id not in ids:
+                ids.append(id)
```

Comment:
**Tip:** We can get rid of this containment check if we use `set()` for `ids` instead of a `list` - we get duplicate detection on new element insertion for free. It's minor though, the code is already readable as is, and its `O(n^2)` complexity probably doesn't hurt here, the expected length of the `ids` list is relatively small, right?